elastic/kibana
# [Index Management] Add deprecation warning to index template and component template UIs

Partially addresses https://github.com/elastic/kibana/issues/170805

## Summary

With https://github.com/elastic/elasticsearch/pull/101148 we now can alert the users when a component template or index template will soon be deprecated and shouldnt be relied upon. This PR adds a badge and a callout to the component template and index template UIs to alert the users about this.

##### How to test

* Start up kibana and es
* Navigate to Stack management -> index management (component_templates|index_templates)
* Verify that:
  * Deprecated resources are hidden by default
  * Upon enabling filtering to see them they should pop up on the table with a badge
  * Verify that deprecated resources are highlighted and when editing, we also show a warning callout.


<details>
<summary>Create deprecated component template</summary>

```
PUT _component_template/template_1
{
  "template": {
    "settings" : {
        "number_of_shards" : 1
    }    
  },
  "deprecated": true
}
```
</details>

<details>
<summary>Screenshots</summary>

// index templates

![Screenshot 2024-01-10 at 15 25 26](https://github.com/elastic/kibana/assets/1191206/ff632464-c4ac-43a2-b89d-be423489a82a)
![Screenshot 2024-01-10 at 15 25 31](https://github.com/elastic/kibana/assets/1191206/bb231c7c-5ae3-466d-9411-3cca842124dc)

![Screenshot 2024-01-08 at 13 52 35](https://github.com/elastic/kibana/assets/1191206/c6724867-3a0e-4998-871b-35fd577689bc)
![Screenshot 2024-01-08 at 14 00 46](https://github.com/elastic/kibana/assets/1191206/0b1001d4-c0bb-49a8-862a-746754e34fe0)

// component templates


![Screenshot 2024-01-10 at 14 45 10](https://github.com/elastic/kibana/assets/1191206/675519b8-80d7-466a-a4ad-185bb7f52651)
![Screenshot 2024-01-10 at 14 45 15](https://github.com/elastic/kibana/assets/1191206/0cbe5704-8549-4812-9202-e44b8f02ac95)
![Screenshot 2024-01-08 at 14 51 03](https://github.com/elastic/kibana/assets/1191206/0288ca12-c8f5-4ed5-ba4c-f77834702e68)
![Screenshot 2024-01-08 at 15 07 17](https://github.com/elastic/kibana/assets/1191206/97f72f52-6816-4bc9-910c-fbe88df50ad2)



</details>
